libensemble/executors/executor.py

```diff
@@ -619,6 +622,8 @@ def polling_loop(
         if calc_status == UNSET_TAG:
             if task.state == "FINISHED":
                 calc_status = WORKER_DONE
+            elif task.state == "FAILED_TO_START":
+                calc_status = TASK_FAILED_TO_START
```

Comment:
In my tests this code is never reached, because when the state is `"FAILED_TO_START"`, `task.finished` is still set to `False`.

This means that the `while` loop will still be executed, and `task.poll` will raise the following exception

```
[0]  2024-02-02 13:30:15,554 libensemble.manager (ERROR): ---- Received error message from worker 3 ----
[0]  2024-02-02 13:30:15,562 libensemble.manager (ERROR): Message: libensemble.executors.executor.ExecutorException: task libe_task_sim_worker3_0 has no process ID - check task has been launched
[0]  2024-02-02 13:30:15,563 libensemble.manager (ERROR): Traceback (most recent call last):
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\worker.py", line 395, in run
    response = self._handle(Work)
               ^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\worker.py", line 354, in _handle
    calc_out, persis_info, calc_status = self._handle_calc(Work, calc_in)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\worker.py", line 272, in _handle_calc
    out = calc(calc_in, Work)
          ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\utils\runners.py", line 51, in run_sim
    return result(calc_in, Work["persis_info"], self.sim_specs, Work["libE_info"], self.sim_f, Work["tag"])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\utils\runners.py", line 95, in _normal_result
    return user_f(*args)
           ^^^^^^^^^^^^^
  File "C:\Users\ferran\Repos\optimas\optimas\sim_functions.py", line 63, in run_template_simulation
    calc_status = execute_and_analyze_simulation(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Repos\optimas\optimas\sim_functions.py", line 118, in execute_and_analyze_simulation
    calc_status = Executor.executor.polling_loop(task, timeout=timeout)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\executors\executor.py", line 606, in polling_loop
    task.poll()
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\executors\executor.py", line 250, in poll
    if not self._check_poll():
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\executors\executor.py", line 226, in _check_poll
    jassert(self.process is not None, f"task {self.name} has no process ID - check task has been launched")
  File "C:\Users\ferran\Miniconda3\envs\optimas_env_py11\Lib\site-packages\libensemble\executors\executor.py", line 107, in jassert
    raise ExecutorException(*args)
libensemble.executors.executor.ExecutorException: task libe_task_sim_worker3_0 has no process ID - check task has been launched
```